processor/src/test/java/org/mapstruct/ap/test/callbacks/ClassContainingCallbacks.java

```diff
@@ -0,0 +1,255 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.callbacks;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.mapstruct.AfterMapping;
+import org.mapstruct.BeforeMapping;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.TargetType;
+
+/**
+ * @author Andreas Gudian
+ */
+public class ClassContainingCallbacks {
+    private static final List<Invocation> INVOCATIONS = new ArrayList<Invocation>();
+
+    @BeforeMapping
+    public void noArgsBeforeMapping() {
+        INVOCATIONS.add( new Invocation( "noArgsBeforeMapping" ) );
+    }
+
+    @BeforeMapping
+    public void withSourceBeforeMapping(Source source) {
+        INVOCATIONS.add( new Invocation( "withSourceBeforeMapping", source ) );
+    }
+
+    @BeforeMapping
+    @Qualified
+    public void withSourceBeforeMappingQualified(Source source) {
```

Comment:
@agudian What is the influence of qualifiers wrt. to before/after mapping methods?